R/utilities-parameter.R

```diff
@@ -46,3 +63,66 @@ toParameters <- function(netParams) {
   sapply(netParams, function(p)
     Parameter$new(p))
 }
+
+#' Set values of parameters
+#'
+#' @param parameters A single object of type 'Parameter' or a list of such objects
+#' The objects are created by calling 'getParameter' or 'getAllParametersMatching'
+#'
+#' @param values A numeric value that should be assigned to the parameter or a vector
+#' of numeric values, if the value of more than one parameter should be changed. Must have the same
+#' length as 'parameters'
+#'
+#' @examples
+#'
+#' #' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' param <- getParameter(c("Organism", "Liver", "Volume"), sim)
+#' setParametersValues(param, 1)
+#' params <- getAllParametersMatching(c("Organism", "Liver", "*", "Volume"), sim)
+#' setParametersValues(params, c(1:6))
+#'
+#' @export
+setParametersValues <- function(parameters, values){
+  singleParameter <- FALSE
+
+  #Test if 'parameter' is an instance of the Parameter-class.
+  if (inherits(parameters, "Parameter") && inherits(parameters, "ObjectBase")){
+    singleParameter <- TRUE
+
+    #Test if parameters and values are of same length
+    if (length(values) != 1){
+      stop(paste0("setParametersValues: 'parameters' and 'Values' must be of the same length!"))
+    }
+  }
+  else{
+    #Test if all parameters are valid R6 objects
+    if (!all(
+      sapply(parameters,
+             fun <- function(x){c(inherits(x, "Parameter"), inherits(x, "ObjectBase"))}))){
+      stop(paste0("setParametersValues: argument 'parameters' is not valid!
+                  Use 'getParameter()' or 'getAllParametersMatching()' to create parameter objects."))
+    }
+
+    #Test if parameters and values are of same length
+    if (length(parameters) != length(values)){
+      stop(paste0("setParametersValues: 'parameters' and 'Values' must be of the same length!"))
+    }
+  }
+
+  #Test if all values are numeric
+  if (!is.numeric(values)){
+    stop(paste0("setParametersValues: argument 'values' is not valid!
+                Must be a vector of numeric entries"))
+  }
+
+  if (singleParameter){
+    parameters$value <- values

```

Comment:
yes. In fact you can simpy do something like `parameters <- c(parameters) and value <-c(values)`
It will convert to an array of one if there is only one element after that. You are good to go and you do not need to check if there is one or more values. The code is the same no matter what